superset/assets/src/visualizations/nvd3_vis.js

```diff
@@ -329,9 +329,9 @@ export default function nvd3Vis(slice, payload) {
             `<tr><td style="color: ${p.color};">` +
               `<strong>${p[fd.entity]}</strong> (${p.group})` +
             '</td></tr>');
-          s += row(fd.x, formatter(p.x));
-          s += row(fd.y, formatter(p.y));
-          s += row(fd.size, formatter(p.size));
+          s += row(fd.x.label, formatter(p.x));
```

Comment:
What about non ad-hoc metric field? seems like we should first test if fd.x is an object or it is just a string. Same thing for the other two.

Reply:
@conglei only ad-hoc metrics are defined for bubble-charts.